src/Features/CSharp/Portable/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseIndexOrRangeOperator
+{
+    using static Helpers;
+
+    /// <summary>
+    /// Analyzer that looks for several variants of code like `s.Slice(start, end - start)` and
+    /// offers to update to `s[start..end]` or `s.Slice(start..end)`.  In order to convert to the
+    /// indexer, the type being called on needs a slice-like method that takes two ints, and returns
+    /// an instance of the same type. It also needs a Length/Count property, as well as an indexer
+    /// that takes a System.Range instance.  In order to convert between methods, there need to be
+    /// two overloads that are equivalent except that one takes two ints, and the other takes a
+    /// System.Range.
+    ///
+    /// It is assumed that if the type follows this shape that it is well behaved and that this
+    /// transformation will preserve semantics.  If this assumption is not good in practice, we
+    /// could always limit the feature to only work on a whitelist of known safe types.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp), Shared]
+    internal partial class CSharpUseRangeOperatorDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
```

Comment:
>AbstractCodeStyleDiagnosticAnalyzer [](start = 70, length = 35)

Same here, should be AbstractBuiltInCodeStyleDiagnosticAnalyzer

Reply:
yup. fixed.